### PR TITLE
disabled student responses from the student test history. If a user m…

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
…anages to access those responses, disable the dropdown

## student test history
screen shot 2018-02-20 at 10 37 48 am

## IGNORE THE ASSESSMENT NAME. If they did manage to access the assessment (manually modifying URL), disable the dropdown
screen shot 2018-02-20 at 10 37 21 am

### CODE REVIEW

**Comment by dsnyder-fairway** [comment]
ignore the assessment names on this pr

File: webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts [hunk 1]
```diff
@@ -67,33 +66,21 @@ export class ResultsByStudentComponent implements OnInit {
...
-  get assessmentType() {
-    return this.assessment.type;
-  }
-
-  get isIab() {
-    return this.assessmentType == AssessmentType.IAB;
-  }
-
-  get isInterim() {
-    return this.assessmentType != AssessmentType.SUMMATIVE;
```
Comment (dsnyder-fairway) [inline webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts]: unrelated changes. Just thought I would clean this up

File: webapp/src/main/webapp/src/app/student/results/student-results.component.html [hunk 2]
```diff
@@ -50,11 +50,12 @@ <h3 class="blue mb-md">{{'labels.student.results.title' | translate}}</h3>
           [student]="examHistory.student"
           [minimumItemDataYear]="minimumItemDataYear"
           [exams]="examsByTypeAndSubject.get(assessmentType).get(subject)"></student-history-iab-table>
-        <student-history-ica-summitive-table
+        <student-history-ica-summative-table
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.html]: 👍

File: webapp/src/main/webapp/src/app/student/responses/student-responses.component.html [hunk 3]
```diff
@@ -28,7 +28,8 @@ <h3 class="blue mb-md">{{'labels.student.responses.title' | translate}}</h3>
                expandableRows="true">
     <p-column field="assessmentItem.position" header="{{'labels.groups.results.assessment.items.cols.number' | translate}}" sortable="true">
       <ng-template let-row="rowData" let-item="rowData.assessmentItem" pTemplate="body">
-        <datatable-row-expander [datatable]="datatable" [row]="row" [text]="item.position"></datatable-row-expander>
+        <datatable-row-expander *ngIf="assessment.isInterim" [datatable]="datatable" [row]="row" [text]="item.position"></datatable-row-expander>
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/responses/student-responses.component.html]: The backend should be fixed to return no item responses for summative exam requests here i think. Basically this page should render nothing if accessed for summative exam IDs

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts [hunk 4]
```diff
@@ -31,11 +36,16 @@ export class StudentHistoryICASummitiveTableComponent implements OnInit {
...
 
-  constructor(private actionBuilder: MenuActionBuilder) {
+  constructor(private actionBuilder: MenuActionBuilder,
+              private translateService: TranslateService) {
   }
 
   ngOnInit(): void {
-    this.actions = this.createActions();
+    if (this.type === AssessmentType.ICA) {
+      this.actions = this.createIcaActions();
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts]: `createInterimActions()` no?

File: webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts [hunk 5]
```diff
@@ -67,33 +66,21 @@ export class ResultsByStudentComponent implements OnInit {
...
 
-  get assessmentType() {
-    return this.assessment.type;
-  }
-
-  get isIab() {
-    return this.assessmentType == AssessmentType.IAB;
-  }
-
-  get isInterim() {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/assessments/results/view/results-by-student/results-by-student.component.ts]: 👍 👍 yay red

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts (hunk 4 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts]: we have an separate `student-history-iab-table` so I think that name may lead to confusion

File: webapp/src/main/webapp/src/app/student/responses/student-responses.component.html (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/responses/student-responses.component.html]: I'll work that in

File: webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts (hunk 4 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/student/results/tables/student-history-ica-summative-table.component.ts]: gotcha. hm we probably shouldn't eventually though. would probably lead to less code if the columns go through some provider interface that accepts assessment type.

File: webapp/src/main/webapp/src/app/student/responses/student-responses.component.html (hunk 3 above)
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/student/responses/student-responses.component.html]: done

File: reporting-service/src/main/resources/application.sql.yml [hunk 6]
```diff
@@ -31,6 +31,7 @@ sql:
       findAllForExam: >-
         ${sql.reporting.snippet.selectFromExamItem}
         WHERE e.id = :exam_id
+        AND a.type_id != '3'
```
Comment (esotrope) [inline reporting-service/src/main/resources/application.sql.yml]: shouldn't be a string right - i think you can drop the `''`

Comment (dsnyder-fairway) [reply reporting-service/src/main/resources/application.sql.yml]: you are right, I removed the `''`

**Comment by esotrope** [comment]
unit test would have caught the '3' thing - let's add a unit test to assert the no-summatives behavior.

**Comment by dsnyder-fairway** [comment]
added an integration test to verify no results found for summative exam with attach exam items. Verified it returns results if we removed the condition where `type != 3`
